pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -100,7 +103,10 @@ func NewCloudProvider(ctx context.Context, options cloudprovider.Options) *Cloud
 		*sess.Config.Region = getRegionFromIMDS(sess)
 	}
 	logging.FromContext(ctx).Debugf("Using AWS region %s", *sess.Config.Region)
-
+	ipv6DNS, err := kubeDNSIPv6(ctx, options.ClientSet)
+	if err != nil {
+		logging.FromContext(ctx).Fatalf("Unable to detect the IP of the kube-dns service, %s", err)
```

Comment:
We should panic here and on EC2 connectivity, IMO. Otherwise we're starting with known permission failures.

Reply:
this is using `Fatalf` to panic, but I can change the ec2 check to panic as well.

Reply:
Oh :doh